yckms/keysource_test.go

```diff
@@ -0,0 +1,184 @@
+package yckms
+
+import (
+	"context"
+	"encoding/base64"
+	yckms "github.com/yandex-cloud/go-genproto/yandex/cloud/kms/v1"
+	kms "github.com/yandex-cloud/go-sdk/gen/kmscrypto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	dummyKey   = "my-dummy-key"
+	dummyKeys  = dummyKey + ",my-dummy-key2"
+	decodedKey = "I want to be a DJ"
+)
+
+const bufSize = 1024 * 1024
+
+var lis *bufconn.Listener
+
+func init() {
+	lis = bufconn.Listen(bufSize)
+	s := grpc.NewServer()
+	yckms.RegisterSymmetricCryptoServiceServer(s, mockSymmetricCryptoServiceServer{})
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			log.Fatalf("Server exited with error: %v", err)
+		}
+	}()
+}
+
+func bufDialer(context.Context, string) (net.Conn, error) {
+	return lis.Dial()
+}
+
+type mockSymmetricCryptoServiceServer struct {
+}
+
+func (mockSymmetricCryptoServiceServer) Encrypt(context.Context, *yckms.SymmetricEncryptRequest) (*yckms.SymmetricEncryptResponse, error) {
+	key, _ := base64.StdEncoding.DecodeString("test")
+	return &yckms.SymmetricEncryptResponse{
+		Ciphertext: key,
+	}, nil
+}
+func (mockSymmetricCryptoServiceServer) Decrypt(context.Context, *yckms.SymmetricDecryptRequest) (*yckms.SymmetricDecryptResponse, error) {
+	return &yckms.SymmetricDecryptResponse{
+		Plaintext: []byte(decodedKey),
+	}, nil
+}
+func (mockSymmetricCryptoServiceServer) ReEncrypt(context.Context, *yckms.SymmetricReEncryptRequest) (*yckms.SymmetricReEncryptResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method ReEncrypt not implemented")
+}
+func (mockSymmetricCryptoServiceServer) GenerateDataKey(context.Context, *yckms.GenerateDataKeyRequest) (*yckms.GenerateDataKeyResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method GenerateDataKey not implemented")
+}
+
+func TestNewMasterKeyFromKeyID(t *testing.T) {
+	key, err := NewMasterKeyFromKeyID(dummyKey)
+	assert.NoError(t, err)
+	assert.Equal(t, dummyKey, key.KeyID)
+	assert.NotNil(t, key.CreationDate)
+}
+
+func TestNewMasterKeyFromKeyIDString(t *testing.T) {
+	keys, err := NewMasterKeyFromKeyIDString(dummyKeys)
+	assert.NoError(t, err)
+	assert.Len(t, keys, 2)
+	assert.Equal(t, dummyKey, keys[0].KeyID)
+}
```

Comment:
Test for a case when we pass keys with spaces near `,`, like: `adfgsr343j, ewqrewr421`

Reply:
added